src/main/scala/com/foursquare/common/thrift/bson/ByteStringBuilder.scala

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 Foursquare Labs Inc. All Rights Reserved.
+
+package com.foursquare.common.thrift.bson
+
+import java.io.{ByteArrayInputStream, InputStream}
+import java.nio.charset.Charset
+import java.lang.reflect.Constructor
+
+private[bson] object ByteStringBuilder {
+  val UTF8_CHARSET = Charset.forName("UTF-8")
+  val ASCII_CHARSET = Charset.forName("US-ASCII")
+  val MaxSize = 16 * 1024 * 1024
+  val StringConstructor: Constructor[String] = {
+    try {
+      val constructor = {
+        classOf[String].getDeclaredConstructors.find(_.toString == "java.lang.String(char[],boolean)").get
+      }
+      constructor.setAccessible(true)
+      constructor.asInstanceOf[Constructor[String]]
+    } catch {
+      case e: Exception =>
+        System.err.println("Error reflecting for String(chars, boolean) private constructor. Will use public constructor.")
+        e.printStackTrace
+        null
+    }
+  }
+
+  val SingleByteStrings: Array[String] = (32 to 127).map(b => new String(Array(b.toChar))).toArray
```

Comment:
I guess the saves you the Byte->Char->Array part, even if the string then comes from the intern pool. And the lookup is faster. So, cool.